### DISCUSSION TITLE
Int16 conversion in decentScale.ts

### DISCUSSION THREAD

**Discussion by banoz** [graphefruit/Beanconqueror#299] [Ideas]
There's an issue with the negative weight readings from the scales. Could you check if this will work better?

`
  private async attachNotification() {
    ble.startNotification(this.device_id, DecentScale.READ_SERVICE_UUID, DecentScale.READ_CHAR_UUID,
      async (_data) => {
        const scaleData = new Uint8Array(_data);
        // console.log("Received: " + scaleData[1] + " - " + scaleData[2] + " - "+ scaleData[3]);
        if (scaleData[1] === 0xCE || scaleData[1] === 0xCA) {
          // Weight notification
          let newWeight: number = scaleData.readInt16BE(2) / 10; // buf.readInt16BE() from node.js

          const weightIsStable = (scaleData[1] === 0xCE);
          this.setWeight(newWeight, weightIsStable);

        } else if (scaleData[1] === 0xAA && scaleData[2] === 0x01) {
          // Tare button pressed.
          this.tareEvent.emit();
          await this.tare();
        } else if (scaleData[1] === 0xAA && scaleData[2] === 0x02) {
          // Timer button pressed
          this.timerEvent.emit(null);
        }
      }, (_data) => {

      }
    );
  }
`

**Comment by graphefruit** [comment]
Hi @banoz,
what is the issue with the actual implementation?
I'm using the decent-scale myself without any issues until now.

**Reply by banoz** [reply]
the app shows -25 something grams if the weight goes below zero after tare command

**Reply by graphefruit** [reply]
Are you using the 1.1 scale?

**Reply by banoz** [reply]
No
